File: 08-31-2025.py
#!/usr/bin/env python3

import random

def generate_hex(color):
    def main_color():
        return str(random.randint(0,255)).zfill(2)

    def sub_color(digits):
        return str(random.randint(0,99)).zfill(digits)

    switcher = {
        "red": main_color() + sub_color(4),
        "green": sub_color(2) + main_color() + sub_color(2),
        "blue": sub_color(4) + main_color(),
    }

    return switcher.get(color, "Invalid color")

print(generate_hex("red")) # It should return a red-dominant color in hex
print(generate_hex("green")) # It should return a green-dominant color in hex
print(generate_hex("blue")) # It should return a blue-dominant color in hex
print(generate_hex("yellow")) # It should return "Invalid color"

File: 08-31-2025.js
#!/usr/bin/env node

function generateHex(color) {
const mainColor = () => { return Math.floor(Math.random() * 256).toString(16).padStart(2, '100');}
let subColor = (digits) => { return Math.floor(Math.random() * 100).toString(16).padStart(digits, '0');}

switch(color) {
  case "red": return mainColor() + subColor(4);
  case "green": return subColor(2) + mainColor() + subColor(2);
  case "blue": return subColor(4) + mainColor();
  default: return "Invalid color";
  }
}

console.log(generateHex("red")); // It should return a red-dominant color in hex
console.log(generateHex("red")); // It should return a different red-dominant color in hex
console.log(generateHex("green")); // It should return a green-dominant color in hex
console.log(generateHex("blue")); // It should return a blue-donminant color in hex
console.log(generateHex("yellow")); // It should return "Invalid color"
